jmclient/jmclient/blockchaininterface.py

```diff
@@ -386,6 +388,84 @@ def estimate_fee_per_kb(self, N):
             return 10000
         return int(Decimal(1e8) * Decimal(estimate))
 
+class BitcoinCoreNoHistoryInterface(BitcoinCoreInterface):
+
+    def __init__(self, jsonRpc, network):
+        super(BitcoinCoreNoHistoryInterface, self).__init__(jsonRpc, network)
+        self.import_addresses_call_count = 0
+        self.wallet_name = None
+        self.scan_result = None
+
+    def import_addresses_if_needed(self, addresses, wallet_name):
+        self.import_addresses_call_count += 1
+        if self.import_addresses_call_count == 1:
+            self.wallet_name = wallet_name
+            addr_list = ["addr(" + a + ")" for a in addresses]
+            log.debug("Starting scan of UTXO set")
+            st = time.time()
+            self.rpc("scantxoutset", ["abort"])
+            self.scan_result = self.rpc("scantxoutset", ["start", addr_list])
```

Comment:
RPC failure must be handled here, as `scantxoutset` was added in Bitcoin Core 0.17, but we still support 0.16.3 too (which is still default stable version on many Linux distros).

Reply:
Done